reward/src/lib.rs

```diff
@@ -0,0 +1,532 @@
+                .iter()
+                .map(|key| -> PeerId { key.to_owned().into() })
+                .collect::<Vec<PeerId>>();
+            if let Ok(Some(ens)) = person.payload().get_ext::<Ens>() {
+                for peer_id in peer_ids {
+                    let fp = keys::to_ssh_fingerprint(&peer_id)?;
+                    ssh_to_ens.insert(fp.to_string(), ens.clone());
+                }
+            }
+        }
+    }
+
+    term::headline("Retrieving commits...");
+    let mut table = term::Table::default();
+    table.push([
+        term::format::bold("Commit"),
+        term::format::bold("Signature"),
+        term::format::bold("ENS"),
+    ]);
+
+    let mut head_sha1 = None;
+    // show commits in provided range and collect all ens which have contributed
+    let all_ens = revwalk
+        .filter_map(|sha1| {
+            let sha1 = match sha1 {
+                Err(_) => return None,
+                Ok(sha1) => sha1,
+            };
+            if head_sha1.is_none() {
+                head_sha1 = Some(sha1);
+            }
+
+            let ens = {
+                let fp = git::commit_ssh_fingerprint(Path::new("."), &sha1.to_string()).ok()?;
+                if fp.is_none() {
+                    table.push([
+                        term::format::highlight(sha1),
+                        term::format::tertiary("missing"),
+                        term::format::secondary("missing"),
+                    ]);
+                    return None;
+                }
+                let fp = fp?;
+                let ens = ssh_to_ens.get(&fp);
+                table.push([
+                    term::format::highlight(sha1),
+                    term::format::tertiary(fp.clone()),
+                    term::format::secondary(
+                        ens.map(|ens| ens.name.to_owned())
+                            .unwrap_or_else(|| String::from("missing")),
+                    ),
+                ]);
+                ens
+            };
+
+            ens
+        })
+        .collect::<Vec<_>>();
+    table.render();
+    term::blank();
+
+    if all_ens.is_empty() {
+        return Err(anyhow!(
+            "No contributor with a set ENS was found in the given range"
+        ));
+    }
+
+    let token: Token = options
+        .token
+        .or_else(|| {
+            term::select_with_prompt(
+                "What's the reward's token?",
+                &["wETH", "DAI", "USDC", "USDT"],
+                &"wETH",
+            )
+            .unwrap()
+            .parse()
+            .ok()
+        })
+        .ok_or_else(|| anyhow!("Couldn't determine reward's token"))?;
+
+    let address = token
+        .get_address()
+        .or_else(|| {
+            term::text_input("Please specify token's address?", Some("".to_string()))
+                .unwrap_or_else(|_| "".to_string())
+                .parse()
+                .ok()
+        })
+        .ok_or_else(|| anyhow!("Couldn't parse token's address"))?;
```

Comment:
Can remove this now, since token always has an address.

Reply:
https://github.com/radicle-dev/radicle-cli/compare/f4007e459f8ce654f3ca58ce48dc1f9910838534..a664622f7600c99dc0622b776816712d0b049d45#diff-26d7e6053383a7793e880ae8b6ee5d2f7d7c23ccd5e88e43f44052374e9f0f58L254-R252